flink/v2.0/flink/src/main/java/org/apache/iceberg/flink/actions/Actions.java

```diff
@@ -32,21 +35,52 @@ public class Actions {
 
   private final StreamExecutionEnvironment env;
   private final Table table;
+  private final TableLoader tableLoader;
 
+  @Deprecated
   private Actions(StreamExecutionEnvironment env, Table table) {
     this.env = env;
     this.table = table;
+    this.tableLoader = null;
   }
 
+  private Actions(StreamExecutionEnvironment env, TableLoader tableLoader) {
+    this.env = env;
+    this.tableLoader = tableLoader;
+    if (!tableLoader.isOpen()) {
+      tableLoader.open();
+    }
+
+    this.table = tableLoader.loadTable();
+  }
+
+  @Deprecated
   public static Actions forTable(StreamExecutionEnvironment env, Table table) {
     return new Actions(env, table);
   }
 
+  public static Actions forTable(StreamExecutionEnvironment env, TableLoader tableLoader) {
+    return new Actions(env, tableLoader);
+  }
+
+  @Deprecated
   public static Actions forTable(Table table) {
     return new Actions(StreamExecutionEnvironment.getExecutionEnvironment(CONFIG), table);
   }
 
+  public static Actions forTableLoad(TableLoader tableLoader) {
+    return new Actions(StreamExecutionEnvironment.getExecutionEnvironment(CONFIG), tableLoader);
+  }
+
   public RewriteDataFilesAction rewriteDataFiles() {
```

Comment:
Shall we deprecate this too? Or we will keep this and somehow migrate to the old one?